### PR TITLE
Young

### PR DESCRIPTION
**Pull request by Hogusong**

### CODE REVIEW

**Comment by Hogusong** [comment]
language browser: refresh after delete, 
songlist browser: confirm before delete'

File: views/songs/language.pug [hunk 1]
```diff
@@ -24,12 +34,46 @@ block content
...
 
         function cancelEdit (code) {
+          permit = true;
           $('#' + code + ' .display').show();
+          $('.add').show();
           $('.editor').hide();
         }
+
+        function deleteItem (code) {
+          // TODO: actural deleting module needed here
```
Comment (chrislim) [inline views/songs/language.pug]: cleanup this comment.

File: views/songs/language.pug [hunk 2]
```diff
@@ -24,12 +34,46 @@ block content
             button.btn.btn-default Add
 
     script(type="text/javascript").
+        var permit = true;
         function editLanguage (code) {
-          $('#' + code + ' .display').hide();
-          $('#' + code + ' .editor').show();
+          if (permit){
```
Comment (chrislim) [inline views/songs/language.pug]: add space after )

File: views/songs/language.pug [hunk 3]
```diff
@@ -24,12 +34,46 @@ block content
...
+          if (confirm('Do you want to delete this language?')) {
+            m.request({
+              method: 'DELETE',
+              url: '/language',
+              data: {
+                code: code
+              }
+            })
+            .then(function(result) {
+              document.getElementById(code).remove()
```
Comment (chrislim) [inline views/songs/language.pug]: add semicolon

File: views/songs/language.pug [hunk 4]
```diff
@@ -24,12 +34,46 @@ block content
...
+            })
+          }
+        }
+
+        $(document).ready(function() {
```
Comment (chrislim) [inline views/songs/language.pug]: add a comment explaining what this is for.

File: views/songs/songlist-db.pug [hunk 5]
```diff
@@ -38,17 +38,19 @@ block content
...
 
-            $('input:checked').each(function(index, value) {
-                var val = $(this).attr('id');
-                console.log(val)
-                console.log($(this));
-                var $thisInput = $(this);
+            if (confirm('Do you want to delete this language?')) {
+                $('input:checked').each(function(index, value) {
+                    var val = $(this).attr('id');
+                    console.log(val)
```
Comment (chrislim) [inline views/songs/songlist-db.pug]: cleanup unused comments

**Pull request merged**